cborg/src/Codec/CBOR/Read.hs

```diff
@@ -923,12 +940,26 @@ go_fast_end !bs (ConsumeBytes k) =
       DecodedToken sz (Fits bstr)   -> k bstr >>= go_fast_end (BS.unsafeDrop sz bs)
       DecodedToken sz (TooLong len) -> return $! SlowConsumeTokenBytes (BS.unsafeDrop sz bs) k len
 
+go_fast_end !bs (ConsumeByteArray k) =
+    case tryConsumeBytes (BS.unsafeHead bs) bs of
+      DecodeFailure                 -> return $! SlowFail bs "expected string"
+      DecodedToken sz (Fits str)    -> k (BA.fromByteString str) >>= go_fast_end (BS.unsafeDrop sz bs)
+      DecodedToken sz (TooLong len) -> return $! SlowConsumeTokenByteArray (BS.unsafeDrop sz bs) k len
+
 go_fast_end !bs (ConsumeString k) =
     case tryConsumeString (BS.unsafeHead bs) bs of
       DecodeFailure                 -> return $! SlowFail bs "expected string"
-      DecodedToken sz (Fits str)    -> k str >>= go_fast_end (BS.unsafeDrop sz bs)
+      DecodedToken sz (Fits str)    -> case T.decodeUtf8' str of
+                                         Right t -> k t >>= go_fast_end (BS.unsafeDrop sz bs)
+                                         Left e  -> return $! SlowFail bs (show e)
       DecodedToken sz (TooLong len) -> return $! SlowConsumeTokenString (BS.unsafeDrop sz bs) k len
 
+go_fast_end !bs (ConsumeUtf8ByteArray k) =
+    case tryConsumeString (BS.unsafeHead bs) bs of
+      DecodeFailure                 -> return $! SlowFail bs "expected string"
+      DecodedToken sz (Fits str)    -> k (BA.fromByteString str) >>= go_fast_end (BS.unsafeDrop sz bs)
```

Comment:
Do we want to worry about doing the conversion (and copying from `ByteString` to `ByteArray`) strictly here? We do it strictly in the slow path but not here in the fast path. One reason to do so would be to not retain references to the input data, without depending on the decoder to actually force the `ByteArray` it gets.

Reply:
Right, I suspect we should force strictly here. I'll make this change and document the behavior.